Automattic/wp-calypso
# Plugins: Reduxify plugin installation and uninstallation

We have several plugins stores, and they're pretty tightly coupled with a lot of functionality. They're also some of the most complex flux stores that remain. In #24180 we're working on reduxifying them step by step. This incremental approach is required because migration in a single PR is not a feasible task.

This PR reduxifies plugin installation and uninstallation (AKA removal) actions. In order to keep the store and only reduxify the flux actions, it updates the redux action creators to also dispatch the corresponding flux actions and updates all the component flux action usage with redux.

As a next step, we can start reduxifying the corresponding flux store functionality for installation and uninstallation.

See #47711 for similar work we did for plugin activation and deactivation and #47742 for updates and autoupdates.

This is part of #24180.

#### Changes proposed in this Pull Request

* Trigger plugin installation and uninstallation flux actions from Redux.
* Use plugin installation and uninstallation Redux actions instead of flux ones.
* Plugins: Delete obsolete plugin installation and uninstallation flux actions.

#### Testing instructions

* Get a Jetpack site with some plugins for testing.
* Verify installing and removing of plugins still works in `/plugins/:plugin/:site`.
* Verify all tests still pass.

#### Notes

* Linting errors are expected - there are a couple of them in a file we touched. I'm not fixing them here to avoid making the PR extra large.
